torch/csrc/api/include/torch/nn/functional/vision.h

```diff
@@ -0,0 +1,47 @@
+#pragma once
+
+#include <torch/types.h>
+
+namespace torch {
+namespace nn {
+namespace functional {
+
+inline Tensor affine_grid(
+    const Tensor& theta,
+    const IntArrayRef& size,
+    bool align_corners = false) {
+  TORCH_CHECK(
+      theta.is_floating_point(), "Expected theta to have floating point type.");
+
+  if (size.size() == 4) {
+    TORCH_CHECK(
+        theta.dim() == 3 && theta.size(1) == 2 && theta.size(2) == 3,
+        "Expected a batch of 2D affine matrices of shape Nx2x3 for size ",
+        size,
+        ". Got: ",
+        theta.sizes());
+  } else if (size.size() == 5) {
+    TORCH_CHECK(
+        theta.dim() == 3 && theta.size(1) == 3 && theta.size(2) == 4,
+        "Expected a batch of 3D affine matrices of shape Nx3x4 for size ",
+        size,
+        ". Got: ",
+        theta.sizes());
+  } else {
+    TORCH_CHECK(
+        false,
+        "affine_grid only supports 4D and 5D sizes, for 2D and 3D affine "
+        "transforms, respectively. Got: ",
+        size);
```

Comment:
nit: it would be awesome to format the error message as
```
TORCH_CHECK(
    false,
    "affine_grid only supports 4D and 5D sizes, ",
    "for 2D and 3D affine transforms, respectively. ",
    "Got size ", size);
```
to match the Python version even better.

Reply:
It would be awesome to add a test case for this error message as well. Thanks a lot for your help @jon-tow!